raiden/network/transport/matrix.py

```diff
@@ -255,34 +263,35 @@ def send_async(
                 'Do not use send_async for {} messages'.format(message.__class__.__name__),
             )
 
-        message_id = message.message_identifier
-        async_result = AsyncResult()
+        self.log.info(
+            'SEND ASYNC',
+            receiver_address=pex(receiver_address),
+            message=message,
+            queue_identifier=queue_identifier,
+        )
+
         if isinstance(message, Processed):
-            async_result.set(True)  # processed messages shouldn't get a Delivered reply
-            self._send_immediate(receiver_address, json.dumps(message.to_dict()))
+            self._send_immediate(queue_identifier, message)
```

Comment:
Because that would require a `Delivered` message for it, and it'd conflict with the `message_id` of the original message (`Processed` have the same identifier as the message it's confirming).